src/NzbDrone.Core/ImportLists/PTP/PTPParser.cs

```diff
@@ -0,0 +1,52 @@
+using System.Collections.Generic;
+using System.Net;
+using Newtonsoft.Json;
+using NzbDrone.Core.ImportLists.Exceptions;
+using NzbDrone.Core.ImportLists.ImportListMovies;
+
+namespace NzbDrone.Core.ImportLists.PTP;
+
+public class PTPParser : IParseImportListResponse
+{
+    public PTPParser()
+    {
+    }
+
+    public IList<ImportListMovie> ParseResponse(ImportListResponse importResponse)
+    {
+        var movies = new List<ImportListMovie>();
+
+        if (!PreProcess(importResponse))
+        {
+            return movies;
+        }
+
+        var response = JsonConvert.DeserializeObject<CollectionPage>(importResponse.HttpResponse.Content);
+
+        if (response == null)
+        {
+            return movies;
+        }
+
+        foreach (var item in response.CoverView.Movies)
+        {
+            movies.Add(new ImportListMovie()
+            {
+                Title = WebUtility.HtmlDecode(item.Title),
+                ImdbId = $"tt{item.ImdbId}"
+            });
+        }
+
+        return movies;
+    }
+
+    protected virtual bool PreProcess(ImportListResponse importListResponse)
+    {
+        if (importListResponse.HttpResponse.StatusCode != HttpStatusCode.OK)
+        {
+            throw new ImportListException(importListResponse, "PTP API call resulted in an unexpected status code ({0})", importListResponse.HttpResponse.StatusCode);
```

Comment:
should probably at least handle a 401 and inform the user the api key is invalid?

Reply:
Perfectly fine with doing this, but the other lists don't do this.

Reply:
trakt is the only other list that requires auth right? so it would just need to be added for to trakt for lists besides ptp

indexers are a different can of worms

Reply:
CouchPotato and Radarr also require auth.